File: src/main/kotlin/firebase/FirebaseConfig.kt
package firebase

import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions

sealed class FirebaseConfig {

    companion object {

        fun photosPath(env: String): String {
            val envPrefix = if(env == "prod") "" else "/${env}"
            return "${envPrefix}/photos/jorbu"
        }

        fun photoPath(env: String, photoId: String): String {
            return "${photosPath(env)}/${photoId}"
        }


        fun trainingPath(person: String): String {
            return "/users/${person}/exercises"
        }

        val firebaseApp: FirebaseApp

        private val serviceAccount = FirebaseConfig::class.java.getResourceAsStream("/.secret/banometer-firebase-adminsdk.json");
        private val options = FirebaseOptions.builder()
            .setCredentials(GoogleCredentials.fromStream(serviceAccount))
            .setDatabaseUrl("https://banometer.firebaseio.com")
            .build();

        init {
            firebaseApp = FirebaseApp.initializeApp(options)
        }

    }
}

File: frontend/src/firebase-storage/firebase-config.ts
import firebase from 'firebase/app';
import 'firebase/database';

const firebaseConfig = {
  apiKey: "AIzaSyA_fTqk6HY8aJdSZUbYOnRUEhBfLkREcKc",
  authDomain: "banometer.firebaseapp.com",
  databaseURL: "https://banometer.firebaseio.com",
  projectId: "banometer",
  storageBucket: "banometer.appspot.com",
  appId: "1:986707920899:web:4c516946699b7cad122636"
};

const firebaseApp = firebase.initializeApp(firebaseConfig);

export const database = firebaseApp.database();
export const photosPath = "/photos/jorbu";
export const trainingPath = (person: string) => {
  return `/users/${person}/exercises`;
};
